src/main/java/com/paraam/cpeagent/core/CPEWorker.java

```diff
@@ -61,15 +61,10 @@ public void run () {
 		//((ConfParameter)confdb.confs.get("InternetGatewayDevice.ManagementServer.PeriodicInformInterval")).value = "1800";
 
         XmlFormatter xmlfmt;
-        
-        if(xmlformatter.equals("normal")) {
-            xmlfmt = new XmlFormatter();
-        }
-        else if(xmlformatter.equals("stripdec")) {
+        if(xmlformatter != null && xmlformatter.equals("stripdec")) {
             xmlfmt = new XmlFormatter(true);
-        }
-        else {
-            xmlfmt = null;
+        } else {
+            xmlfmt = new XmlFormatter();
```

Comment:
Default case should be not to apply formatting.
Should be
```java
        if(xmlformatter != null && xmlformatter.equals("stripdec")) {
            xmlfmt = new XmlFormatter(true);
        } else if(xmlformatter != null && xmlformatter.equals("normal")) {
            xmlfmt = new XmlFormatter();
        } else {
            xmlfmt = null;
        }
```

I think you were getting crashes because of `xmlformatter.equals` being called on a null String object

Reply:
The point is that also going over with `xmlfmt==null` (using the CPEMain entrypoint https://github.com/LuKePicci/tr069-simulator/blob/master/src/main/java/com/paraam/cpeagent/core/CPEMain.java#L15 for example) will crash in the costructor right after
https://github.com/LuKePicci/tr069-simulator/pull/2/files/661689851a0bb3eb40dba2c4ea821ad74664cbac#diff-78d430c84488d8e75099460baf4637dfR70

so seems a formatter is needed

Reply:
There is an issue with some entry points I never tested where default values are null instead of empty strings. I have just cleaned up the xml formatting setup implementation from some leftovers. Let me push these and other changes directly into master branch..